deeplinkdispatch-base/src/main/java/com/airbnb/deeplinkdispatch/Parser.java

```diff
@@ -1,8 +1,40 @@
 package com.airbnb.deeplinkdispatch;
 
+import java.util.List;
+
 /**
  * Parser classes are used for keeping a registry of deep links and dispatching them from Intents.
  */
-public interface Parser {
-  DeepLinkEntry parseUri(String uri);
+public abstract class Parser {
+
+  public Parser(List<DeepLinkEntry> registry) {
+    this.registry = registry;
+  }
+
+  /**
+   * The List of DeepLinkEntries collected from a Module. Annotation Processing-generated Children
+   * of this class will pass the collected {@code List<DeepLinkEntry>} in the constructor's
+   * {@code super()}. The generated implementation will be a {@link
+   * Collections#unmodifiableList(java.util.List)}.
+   */
+  private final List<DeepLinkEntry> registry;
+
+  /**
+   * Notice that {@link com.airbnb.deeplinkdispatch.Parser#registry} is immutable.
+   * @return {@code Collections.unmodifiableList<DeepLinkEntry>} {@link Parser#registry}
+   */
+  public List<DeepLinkEntry> getRegisteredDeepLinks() {
+    return registry;
+  }
+
+  public DeepLinkEntry parseUri(String uri) {
```

Comment:
Some thoughts:

- This seems like something we should address when [generating the list](com.airbnb.deeplinkdispatch.DeepLinkProcessor#generateDeepLinkLoader), not when retrieving elements from it. 
  - If we wanted to prevent **intra**-module duplicates we could fail at compile-time by adding some checks, using an immutable Set (instead of list, `Collections.unmodifiableSet`), or both.
  - I'm not sure if app-wide duplicates are a possibility, but we could also check for this.
- This doesn't change the prior behavior, which would have generated code identical code like in the [tests](https://github.com/airbnb/DeepLinkDispatch/pull/260/files/7085625781a2b64e8b1c429e96d40d7aa265bf5f#diff-49396e9f26b9dc5c42b6266107b3832aL75).
- I don't want to tackle additional scope in this PR.

Reply:
Maybe we should create a ticket . to track this?

Reply:
https://github.com/airbnb/DeepLinkDispatch/issues/261